src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file preprocess_imputer_main.cpp
+ * @author Keon Kim
+ *
+ * a utility that provides imputation strategies fore
+ * missing values.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/imputer.hpp>
+#include <mlpack/core/data/dataset_info.hpp>
+#include <mlpack/core/data/map_policies/increment_policy.hpp>
+#include <mlpack/core/data/map_policies/missing_policy.hpp>
+#include <mlpack/core/data/imputation_methods/mean_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/median_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/custom_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/listwise_deletion.hpp>
+
+PROGRAM_INFO("Impute Data", "This utility takes a dataset and converts user "
+    "defined missing variable to another to provide more meaningful analysis "
+    "\n\n"
+    "The program does not modify the original file, but instead makes a "
+    "separate file to save the output data; The program requires you to "
+    "specify the file name with --output_file (-o)."
+    "\n\n"
+    "For example, if we consider 'NULL' in dimension 0 to be a missing "
+    "variable and want to delete whole row containing the NULL in the "
+    "column-wise dataset, and save the result to result.csv, we could run"
+    "\n\n"
+    "$ mlpack_preprocess_imputer -i dataset.csv -o result.csv -m NULL -d 0 \n"
+    "> -s listwise_deletion")
+
+PARAM_STRING_REQ("input_file", "File containing data,", "i");
+PARAM_STRING("output_file", "File to save output", "o", "");
+PARAM_STRING("missing_value", "User defined missing value", "m", "")
+PARAM_STRING("strategy", "imputation strategy to be applied", "s", "")
+PARAM_DOUBLE("custom_value", "user_defined custom value", "c", 0.0)
+PARAM_INT("dimension", "the dimension to apply imputation", "d", 0);
```

Comment:
Updated so that when --dimension is not specified, the program applies the imputation strategy to all dimensions.